Automattic/woocommerce-payments
# Add Tracking Property for New Onboarding Flow

Fixes #6710 

#### Changes proposed in this Pull Request



Adds a new property, `is_new_onboarding` flow to the `wcpay_connect_account_clicked` event so that we better segment our tracking data for non-new UX vs. new UX for onboarding.







#### Testing instructions





* Pull this branch and build the JS.
* Install the [Tracks Vigilante](https://github.com/Automattic/tracks-chrome-extension) Chrome extension which will help a lot in testing this PR. (there's also [Tracks Vigilante 2](https://github.com/Automattic/tracks-vigilante-2-chrome-extension), which I haven't tested yet but has more features).
* On your local server, go to `localhost:8087` and edit the `wcpay_accounts` table to change the `account_type` value for any accounts to something other than `wcpay` (I use `wcpay_old` for example). This forces the client to not have a connected account.
* You will probably need to go to WooCommerce > Settings > Advanced > WooCommerce.com and set `Allow usage of WooCommerce to be Tracked`  to enabled.
* Enable `Progressive Onboarding` in the WCPay dev tools.
* Open Tracks Vigilante (you can open a standalone window). 
* Go to `Payments` and click `Finish Setup`. 
* You should see the `wcadmin_wcpay_connect_account_clicked` event appear in Tracks Vigilante.
* Check that the `is_new_onboarding_flow` parameter is present and the value is true.
* Go back, go to the dev tools and turn `Progressive Onboarding off.
* Repeat the process. This time, the `is_new_onboarding_flow` parameter should be false.
* Clear the account again from your WCPay local server (if it has been created), and refresh your account cache on the client.
* Go to the experiment in Abacus (search MGS for `Progressive Onboarding Experiment` to find it. Use the provided bookmarklet for the treatment group, add it to your bookmarks and click on it with your store open and your browser web tools console open. 
* Start the onboarding again, the event should appear in Tracks Vigilante and the value of `is_new_onboarding_flow` should be true.

-------------------


**Post merge**
